projects/golang/Dockerfile

```diff
@@ -0,0 +1,24 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+
+FROM gcr.io/oss-fuzz-base/base-builder
+MAINTAINER guidovranken@gmail.com
+
+RUN apt-get update && apt-get install -y build-essential wget
+RUN wget https://dl.google.com/go/go1.11.1.linux-amd64.tar.gz
```

Comment:
Most of go-fuzz-corpus tests the standard library, so if we download the fixed revision, we test the fixed revision.
I would assume that all existing fuzzers should checkout a git with the project to get latest sources for testing?

Reply:
And FWIW 1.12 is already out.

Reply:
For some reason I get an import cycle error when I run it on 1.12. Any idea why? To test s/1.11.1/1.12/g in Dockerfile and build.

Reply:
What import cycle? On what binary? What is "build"?

Reply:
```
failed to execute go build: exit status 1
import cycle not allowed
package github.com/dvyukov/go-fuzz-corpus/url/go.fuzz.main
	imports github.com/dvyukov/go-fuzz-corpus/url
	imports fmt
	imports go-fuzz-dep
	imports syscall
	imports runtime
	imports runtime/internal/math
	imports go-fuzz-dep
```

Happens for 1.12 but not for 1.11.1.

To test:

```
git clone https://github.com/google/oss-fuzz
cd oss-fuzz
[change downloaded Go version from 1.11.1 to 1.12 in projects/golang/Dockerfile]
python infra/helper.py build_fuzzers golang
```

Reply:
Adding  runtime/internal/math here should help:
https://github.com/dvyukov/go-fuzz/blob/master/go-fuzz-build/main.go#L302-L304

Reply:
That appears to fix it, thanks!

With regards to using the latest Go version, does the project offer nightly binaries? If not I'll look into https://golang.org/doc/install/source and try if I get build the latest version from the repo (I've never done that before).

Reply:
Do you mind sending PR for go-fuzz with the runtime/internal/math change? 

Go does not offer nightly releases, but it offer simple and working build :)
git clone; cd go/src; ./make.bash

Reply:
I’m actively updating Go-fuzz-build to work with go/packages, which should fix this issue and similar ones in the future. I hope to send a PR in the next few days. I’d love help testing and reviewing it—it ends up being a pretty major rewrite.

Reply:
> Do you mind sending PR for go-fuzz with the runtime/internal/math change?

https://github.com/dvyukov/go-fuzz/pull/211 should fix this, in a much deeper way. Still needs testing (and probably fixes) on Windows. Help there--even just bug reports--would be deeply appreciated.

Reviews from any/all are also very welcome. It's a large change.